src/amberscript/parser.cc

```diff
@@ -1066,8 +1066,42 @@ Result Parser::ParseExpect() {
     return Result("unknown buffer name for EXPECT command");
 
   token = tokenizer_->NextToken();
-  if (!token->IsString() || token->AsString() != "IDX")
-    return Result("missing IDX in EXPECT command");
+
+  if (!token->IsString())
+    return Result("Invalid comparator in EXPECT command");
+
+  if (token->AsString() == "EQ_BUFFER") {
+    token = tokenizer_->NextToken();
+    if (!token->IsString())
+      return Result("invalid buffer name in EXPECT EQ_BUFFER command");
+
+    auto* buffer_2 = script_->GetBuffer(token->AsString());
+    if (!buffer_2)
+      return Result("unknown buffer name for EXPECT EQ_BUFFER command");
```

Comment:
Add the buffer name?
```
("unknown buffer name for EXPECT EQ_BUFFER command: " + token->AsString())
```

Reply:
Good point, I also added name of unknown buffer if the first buffer argument is unknown